src/certificate.rs

```diff
@@ -400,3 +401,49 @@ mod read_pki {
         }
     }
 }
+
+///Write information about certificate found in slot a la yubico-piv-tool output.
+pub fn print_cert_info(yubikey: &mut YubiKey, slot: SlotId) -> Result<(), Error> {
+    let txn = yubikey.begin_transaction()?;
+    let buf = match read_certificate(&txn, slot) {
+        Ok(b) => b,
+        Err(e) => {
+            println!("error reading certificate in slot {:?}: {}", slot, e);
+            return Err(e);
+        }
+    };
+
+    if !buf.is_empty() {
+        let mut hasher = Sha256::new();
+        hasher.input(buf.clone().to_vec());
+        let fingerprint = hasher.result();
```

Comment:
You can accomplish the above with:

```suggestion
        let fingerprint = Sha256::digest(buf);
```

Reply:
this worked: let fingerprint = Sha256::digest(&buf);